seatunnel-connectors-v2/connector-kudu/src/main/java/org/apache/seatunnel/connectors/seatunnel/kudu/source/KuduSource.java

```diff
@@ -0,0 +1,179 @@
+            SourceSplitEnumerator.Context<KuduSourceSplit> enumeratorContext) {
+        return new KuduSourceSplitEnumerator(enumeratorContext,partitionParameter);
+    }
+
+    @Override
+    public SourceSplitEnumerator<KuduSourceSplit, KuduSinkState> restoreEnumerator(
+            SourceSplitEnumerator.Context<KuduSourceSplit> enumeratorContext, KuduSinkState checkpointState) {
+        // todo:
+        return new KuduSourceSplitEnumerator(enumeratorContext,partitionParameter);
+    }
+
+    @Override
+    public Serializer<KuduSinkState> getEnumeratorStateSerializer() {
+        return new DefaultSerializer<>();
+    }
+
+    @Override
+    public String getPluginName() {
+        return "KuduSource";
+    }
+
+    @Override
+    public void prepare(Config config) {
+
+        String kudumaster = config.getString(KuduSourceConfig.kuduMaster);
+        String tableName = config.getString(KuduSourceConfig.tableName);
+
+        String columnslist = config.getString(KuduSourceConfig.columnsList);
+
+
+        kuduInputFormat=new KuduInputFormat(kudumaster,tableName,columnslist);
+        try {
+            KuduClient.KuduClientBuilder kuduClientBuilder = new
+                    KuduClient.KuduClientBuilder(kudumaster);
+            kuduClientBuilder.defaultOperationTimeoutMs(1800000);
+
+            KuduClient kuduClient = kuduClientBuilder.build();
+            partitionParameter = initPartitionParameter(kuduClient,tableName);
+            SeaTunnelRowType seaTunnelRowType =getSeaTunnelRowType(kuduClient.openTable(tableName).getSchema().getColumns());
+
+
+            rowTypeInfo=seaTunnelRowType;
+        } catch (KuduException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private PartitionParameter initPartitionParameter(KuduClient kuduClient,String tableName) {
+        String keyColumn = null;
+        ArrayList<Integer> keyList = null;
+        try {
+            KuduScanner.KuduScannerBuilder kuduScannerBuilder =
+                    kuduClient.newScannerBuilder(kuduClient.openTable(tableName));
+            ArrayList<String> columnsList = new ArrayList<String>();
+            keyColumn = kuduClient.openTable(tableName).getSchema().getPrimaryKeyColumns().get(0).getName();
+            columnsList.add(""+keyColumn);
+            kuduScannerBuilder.setProjectedColumnNames(columnsList);
+
+            KuduScanner kuduScanner = kuduScannerBuilder.build();
+            keyList = new ArrayList<>();
+            while (kuduScanner.hasMoreRows()) {
+                RowResultIterator rowResults = kuduScanner.nextRows();
+                while (rowResults.hasNext()) {
+                    RowResult row = rowResults.next();
+                    int id = row.getInt(""+keyColumn);
+                    keyList.add(id);
+
+                }
+
+            }
+        } catch (KuduException e) {
+            e.printStackTrace();
+        }
+        Collections.sort(keyList);
```

Comment:
If you only need know the minimum and maximum value of the keyCloumn, you need not add all values in the list. You can find the maximum and minimum values by comparing in the while loop.

Reply:
ok，thx